src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1190,11 +1190,7 @@ void SolverInterfaceImpl:: writeVectorData
 # endif
   CHECK(valueIndex >= -1, "Invalid value index (" << valueIndex << ") when writing vector data!" );
   if (_clientMode){
-    Eigen::VectorXd valueCopy(_dimensions);
-    for (int dim=0; dim < _dimensions; dim++){
-      valueCopy[dim] = value[dim];
-    }
-    _requestManager->requestWriteVectorData(fromDataID, valueIndex, valueCopy.data());
```

Comment:
@floli Can we really delete this block of code? To me it looks fine to remove it, but there might be a reason for the existence of these lines that I do not see...

Reply:
This change doesn't really fit in with this PR, so I would vouche to fix this in the future.

Reply:
I reverted the changes in https://github.com/precice/precice/pull/348/commits/426f49e6c4fcdd170b5c19835efd3b420904ea7f. We can deal with this in #356